### PR TITLE
Source-acceptance-test: improvements

### LINKED ISSUE #3467: SAT: Usability improvements
## Tell us about the problem you're trying to solve
Currently, we have few issues with the tests:

* [ ] If `cursor_field` is None then `default_cursor_field` should be used instead
* [ ] It should be possible to skip `cursor_paths` if they equal to the last item in paths in records (default behavior in test fixtures).
* [ ] rename `incremental.state_path` -> `incremental.abnormal_state_path`

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368072883580) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by vovavovavovavova**
## How
By solving listed issues described in the issue.
Please note, that solving (3) issue makes old `acceptance-test-config.yml` files incompatible, because it contains old `state_path` field and does not contain required `future_state_path`. Config field need to be overwritten.

### CODE REVIEW

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py [hunk 1]
```diff
@@ -51,11 +51,12 @@ def future_state_fixture(future_state_path) -> Path:
 
 @pytest.fixture(name="cursor_paths")
 def cursor_paths_fixture(inputs, configured_catalog_for_incremental) -> Mapping[str, Any]:
-    cursor_paths = getattr(inputs, "cursor_paths")
+    cursor_paths = getattr(inputs, "cursor_paths", None)
```
Comment (keu) [inline airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py:54]:
```suggestion
    cursor_paths = getattr(inputs, "cursor_paths", {})
```

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py [hunk 2]
```diff
@@ -51,11 +51,12 @@ def future_state_fixture(future_state_path) -> Path:
...
     result = {}
 
     for stream in configured_catalog_for_incremental.streams:
-        path = cursor_paths.get(stream.stream.name, [stream.cursor_field[-1]])
+        default_ = [stream.cursor_field[-1]]
```
Comment (keu) [inline airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py:58]: then you don't need to change anythin here

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py [hunk 3]
```diff
@@ -66,7 +66,13 @@ def configured_catalog_for_incremental_fixture(configured_catalog) -> Configured
     catalog = incremental_only_catalog(configured_catalog)
     for stream in catalog.streams:
         if not stream.cursor_field:
-            pytest.fail("Configured catalog should have cursor_field specified for all incremental streams")
+            if stream.stream.default_cursor_field:
+                stream.cursor_field = stream.stream.default_cursor_field[:]
+            else:
+                pytest.fail(
+                    "Configured catalog should have `cursor_field` " "either `default_cursor_field` specified for all incremental streams"
```
Comment (sherifnada) [inline airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py:73]: suggested wording improvement
```suggestion
                    f"All incremental streams should either have `cursor_field` declared in the configured catalog or `default_cursor_field` specified in the catalog output by discover. Stream {stream.stream.name} does not have either property defined."
```

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py [hunk 4]
```diff
@@ -51,7 +51,7 @@ def future_state_fixture(future_state_path) -> Path:
 
 @pytest.fixture(name="cursor_paths")
 def cursor_paths_fixture(inputs, configured_catalog_for_incremental) -> Mapping[str, Any]:
-    cursor_paths = getattr(inputs, "cursor_paths")
+    cursor_paths = getattr(inputs, "cursor_paths") or {}
```
Comment (sherifnada) [inline airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py:54]: what is the impact of this change?

**Review by sherifnada — APPROVED** [review]
LGTM with some comments. 

Can you create a follow up PR which renames `state_path` in all connectors? 

Also you will need to publish the SAT docker image

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py (hunk 4 above)
Comment (vovavovavovavova) [reply airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py:54]: cursor_paths = getattr(inputs, "cursor_paths") returns None if cursor_paths are commented in yaml, and this case fail few lines below on cursor_paths.get

**Review by sherifnada — APPROVED** [review]

**Pull request merged**